lightning/src/ln/channel.rs

```diff
@@ -6918,26 +7097,38 @@ impl<SP: Deref> FundedChannel<SP> where
 		Ok((closing_signed, None, None))
 	}
 
-	// Marks a channel as waiting for a response from the counterparty. If it's not received
-	// [`DISCONNECT_PEER_AWAITING_RESPONSE_TICKS`] after sending our own to them, then we'll attempt
-	// a reconnection.
-	fn mark_awaiting_response(&mut self) {
-		self.context.sent_message_awaiting_response = Some(0);
+	fn mark_response_received(&mut self) {
+		self.context.sent_message_awaiting_response = None;
 	}
 
 	/// Determines whether we should disconnect the counterparty due to not receiving a response
 	/// within our expected timeframe.
 	///
-	/// This should be called on every [`super::channelmanager::ChannelManager::timer_tick_occurred`].
+	/// This should be called for peers with an active socket on every
+	/// [`super::channelmanager::ChannelManager::timer_tick_occurred`].
 	pub fn should_disconnect_peer_awaiting_response(&mut self) -> bool {
-		let ticks_elapsed = if let Some(ticks_elapsed) = self.context.sent_message_awaiting_response.as_mut() {
-			ticks_elapsed
+		if let Some(ticks_elapsed) = self.context.sent_message_awaiting_response.as_mut() {
+			*ticks_elapsed += 1;
+			*ticks_elapsed >= DISCONNECT_PEER_AWAITING_RESPONSE_TICKS
+		} else if
+			// Cleared upon receiving `channel_reestablish`.
+			self.context.channel_state.is_peer_disconnected()
+			// Cleared upon receiving `stfu`.
+			|| self.context.channel_state.is_local_stfu_sent()
+			// Cleared upon receiving a message that triggers the end of quiescence.
+			|| matches!(self.context.channel_state, ChannelState::ChannelQuiescent)
+			// Cleared upon receiving `revoke_and_ack`.
+			|| self.context.has_pending_channel_update(false)
+			|| self.context.has_pending_channel_update(true)
+		{
+			// This is the first tick we've seen after expecting to make forward progress.
+			self.context.sent_message_awaiting_response = Some(1);
+			debug_assert!(DISCONNECT_PEER_AWAITING_RESPONSE_TICKS > 1);
```

Comment:
ops I forget the comment here!

This could be removed right? or we need to explicitly silent le linter

```
error: `debug_assert!(true)` will be optimized out by the compiler
    --> lightning/src/ln/channel.rs:7126:4
     |
7126 |             debug_assert!(DISCONNECT_PEER_AWAITING_RESPONSE_TICKS > 1);
     |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
     |
     = help: remove it
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#assertions_on_constants
     = note: `-D clippy::assertions-on-constants` implied by `-D warnings`
     = help: to override `-D warnings` add `#[allow(clippy::assertions_on_constants)]`
```